### PR TITLE
Fix resources requests for AWS Batch plugin

### PR DESCRIPTION
**Pull request by hamersaw**
# TL;DR
This PR uses the container resources as the base before merging with resource overrides and platform defaults. The logic involved in [this PR](https://github.com/flyteorg/flytepropeller/pull/544) updated always setting the resource overrides even if there were none (if missing it would set overrides to the container resources), which introduced this regression.

## Type
 - [x] Bug Fix
 - [ ] Feature
 - [ ] Plugin

## Are all requirements met?

 - [x] Code completed
 - [x] Smoke tested
 - [ ] Unit tests added
 - [x] Code documentation added
 - [ ] Any pending items have an associated Issue

## Complete description
^^^

## Tracking Issue
https://github.com/flyteorg/flyte/issues/4003

## Follow-up issue
_NA_

### CODE REVIEW

**Review by pingsutw — APPROVED** [review]

**Review by eapolinario — APPROVED** [review]
Does https://github.com/flyteorg/flytepropeller/pull/544 also affect other plugins?

**Comment by hamersaw** [comment]
> Does [flyteorg/flytepropeller#544](https://github.com/flyteorg/flytepropeller/pull/544) also affect other plugins?

Good question, as far as I can tell no. Other plugins use the container parsing we define. AWS Batch is unique because it uses the resources to populate a different object (ie. aws batch job).
